### PR TITLE
added validations and has many relationships to model files

### PR DESCRIPTION
**Pull request by brittanyrjones**
Anything you want checked?

### CODE REVIEW

File: app/models/product.rb [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+  validates :price, presence: true, numericality: {
+    greater_than: 0
+  }
+
+  validates :stock, presence: true, numericality: {
+    greater_than_or_equal_to: 0,
+    only_integer: true
+  }
 
-  def self.status_change #change availability
```
Comment (Oh-KPond) [inline app/models/product.rb]: Did you mean to delete this?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :products
+  has_many :orders_products, through: :products
+  has_many :reviews, through: :products
+  has_many :categories_products, through: :products
```
Comment (Oh-KPond) [inline app/models/user.rb]: I'm not sure you need this one  ?shrug

File: coverage/.resultset.json [hunk 3]
```diff
@@ -1,6 +1,33 @@
 {
```
Comment (Oh-KPond) [inline coverage/.resultset.json]: Hahahaha I don't know what's happening here, but I think it'll be solved when we get everything merged and Steffany does whatever it was that she was going to do.

File: db/schema.rb [hunk 4]
```diff
@@ -58,12 +58,6 @@
     t.integer "user_id"
   end
 
-  create_table "products_categories", force: :cascade do |t|
```
Comment (Oh-KPond) [inline db/schema.rb]: I think getting rid of this is fine, so I'm not going to worry about it

**Comment by Oh-KPond** [comment]
This was my first time using review, so let me know if you have questions! 😄

File: app/models/product.rb (hunk 1 above)
Comment (brittanyrjones) [reply app/models/product.rb]: yes

**Comment by Oh-KPond** [comment]
Looks good! Thanks Brittany!

**Pull request merged**